packages/extension/src/ui/features/actions/ApproveTransactionScreen.tsx

```diff
@@ -74,12 +78,85 @@ export interface ApproveTransactionScreenProps
   ) => void
 }
 
+async function addTokenToBalances(balances: Map<string, BigNumber>, tokenId: string, amount: BigNumber) {
+  const tokenBalance = balances.get(tokenId)
+  if (tokenBalance === undefined) {
+    balances.set(tokenId, amount)
+  } else {
+    balances.set(tokenId, tokenBalance.add(amount))
+  }
+}
+
+async function getBalances(nodeProvider: NodeProvider, address: string): Promise<Map<string, BigNumber>> {
+  const result = await nodeProvider.addresses.getAddressesAddressBalance(address)
+  const balances = new Map<string, BigNumber>()
+  balances.set(ALPH_TOKEN_ID, BigNumber.from(result.balance))
+  if (result.tokenBalances !== undefined) {
+    result.tokenBalances.forEach((token) => addTokenToBalances(balances, token.id, BigNumber.from(token.amount)) )
```

Comment:
Currently the full node only returns once, but in order to prevent future changes to the implementation of this endpoint, I added the `addTokensToBalances` to handle the scenario where the same token is returned multiple times.